packages/next-auth/src/providers/oauth.ts

```diff
@@ -112,8 +98,8 @@ export interface OAuthConfig<P> extends CommonProviderOptions, PartialIssuer {
   version?: string
   profile?: (profile: P, tokens: TokenSet) => Awaitable<User & { id: string }>
   checks?: ChecksType | ChecksType[]
-  client?: Partial<ClientMetadata>
-  jwks?: { keys: JWK[] }
```

Comment:
We might not need this anymore, as the new [Client](https://github.com/panva/oauth4webapi/blob/main/docs/interfaces/Client.md) doesn't require it.

Reply:
This exists to be able to override the usage of `openid-client`. I wonder if we can truly get rid of it and still be flexible. We should discuss.

Reply:
Example: https://github.com/nextauthjs/next-auth/pull/4286

How could we solve it otherwise?